sdk/core/Azure.Core/src/Shared/ConnectionString.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Core
+{
+    internal sealed class ConnectionString
+    {
+        private readonly Dictionary<string, string> _segments;
+        private readonly string _segmentSeparator;
+        private readonly string _keyValueSeparator;
+
+        public static ConnectionString Parse(string connectionString, string segmentSeparator = ";", string keyValueSeparator = "=", bool allowFlags = false, bool allowEmptyValues = false, bool allowWhitespaces = false)
```

Comment:
What is a flag in this context?

Reply:
Something like `url=http://my.azure.com;action=login;ignoreSavedData`

Reply:
I'm still not sure I understand what a flag is, and what it would mean to not allow flags.

Reply:
Removed